lib/tapioca/compilers/symbol_table/symbol_generator.rb

```diff
@@ -321,7 +321,9 @@ def compile_methods(name, constant)
           )
 
           instance_methods = compile_directly_owned_methods(name, constant)
-          singleton_methods = compile_directly_owned_methods(name, constant.singleton_class, [:public])
+
+          constant_singleton = Object.instance_method(:singleton_class).bind(constant).call
```

Comment:
There are a couple of other `singleton_class` uses in this same file, I wonder if we should fix them too?

Reply:
@Morriar I have just spotted those! Working on it now. Thank you.

Reply:
Thanks, good catch. While you are working through the other instances, can you wrap this in a method named `singleton_class_of` like how we've done with similar concerns (like `name_of`, `constants_of` etc at the end of the same file)?

Reply:
Actually I went ahead a did this, which also fixed @Morriar's added test.